ts/src/main/scala/ch/epfl/ts/evaluation/Evaluator.scala

```diff
@@ -0,0 +1,135 @@
+package ch.epfl.ts.evaluation
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import scala.concurrent.duration.FiniteDuration
+import akka.actor.{ActorRef, Cancellable}
+import ch.epfl.ts.component.{ComponentRegistration, Component, ComponentRef, StartSignal, StopSignal}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Evaluates the performance of traders by total returns, volatility, draw down and sharp
+  *
+  * To use this class, redirect all previous connections into and out of the trader to
+  * instances of this class.
+  *
+  * @param trader the reference to the trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the initial seed money
+  * @param period the time period to compute performance
+  */
+class Evaluator(trader: ComponentRef, traderId: Long, initial: Double, currency: Currency, period: FiniteDuration) extends Component {
+  // for usage of scheduler
+  import context._
+
+  private var schedule: Cancellable = null
+  private val wallet = MMap[Currency, Double](currency -> initial)
+  private val returnsList = MList[Double]()
+
+  private var lastValue = initial
+  private var maxProfit = 0.0
+  private var maxLoss = 0.0
+
+  def totalReturns: Double = value() / initial
+  def volatility: Double = computeVolatility
+  def drawdown: Double = maxLoss
+  def sharpeRatio: Double = totalReturns / volatility
+
+  /**
+   * Redirects out-going connections to the trader
+   * */
+  override def connect(ar: ActorRef, ct: Class[_], name: String) = {
+    trader.ar ! ComponentRegistration(ar, ct, name)
+  }
+
+  /**
+   * Handles interested messages and forward all messages to the trader
+   */
+  override def receiver = {
+    case t: Transaction if t.buyerId == traderId =>  // buy
+      trader.ar ! t
+      buy(t)
+    case t: Transaction if t.sellerId == traderId =>  // sell
+      trader.ar ! t
+      sell(t)
+    case 'UpdateStatistics =>
+      updateStatistic
+    case m => trader.ar ! m
+  }
+
+  /**
+   *  Returns the exchange ratio between two currency
+   * */
+  private def ratio(from: Currency, to: Currency): Double = ???
```

Comment:
For now, we don't have a central authority for the current price of currencies except for the Fetchers which receive Quotes directly. If your component is connected to the Fetcher, you will receive quotes and can compute the wallet's total value.
Otherwise, we may have a 'central authority' responsible for holding the latest price, and be able to query it to retrieve the prices. It sounds like these additional messages are not necessary though, since they would always be based on the same `Quote` information that your component could have received. Does it make sense?